### PR TITLE
TDP: Implemented Feature flags for each TDP component

### PR DESCRIPTION
**Pull request by atuggle**
Teachers Digital Platform has multiple sub tools they need to individually turn on and off.  This request adds 3 new Flags and uses 2 of them and we will be using the other in the near future.

## Additions

- Three New Feature Flags:
  - TDP_CRTOOL
    - Used for the Customer Review Tool (turn on and off)
  - TDP_CRTOOL_PROTOTYPES
    - Used for the prototype pages for the Customer Review Tool (turn on and off)
  - TDP_SEARCH_INTERFACE
    - Will be used for the Search Interface we are starting to develop right now

## Removals

- N/A

## Changes

- N/A

## Reviewers
- @hillaryj @Scotchester 

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: cfgov/cfgov/settings/base.py [hunk 1]
```diff
@@ -657,6 +657,15 @@
...
 
+    # Teacher's Digital Platform Customer Review Tool
+    'TDP_CRTOOL': {},
+
+    # Teacher's Digital Platform Customer Review Tool Prototypes Pages
+    'TDP_CRTOOL_PROTOTYPES': {},
+
+    # Teacher's Digital Platform Search Interface Tool
+    'TDP_SEARCH_INTERFACE': {},
+
```
Comment (Scotchester) [inline cfgov/cfgov/settings/base.py]: If these are entirely replacing the existing `TDP_RELEASE` flag, could you please delete that original flag above? (Lines 651-654.)

If you want any of them to be turned on permanently on Beta, you should copy `TDP_RELEASE`'s hardcoded `site` condition.

Comment (atuggle) [reply cfgov/cfgov/settings/base.py]: Nice, not sure how I missed that!  I will remove the TDP_RELEASE, and turn on the Beta flag for them.

**Comment by atuggle** [comment]
@Scotchester Removed the TDP_RELEASE flag as it is not needed at this time.  Added beta site flag to the other flags.

File: cfgov/cfgov/urls.py [hunk 2]
```diff
@@ -394,17 +394,17 @@ def flagged_wagtail_template_view(flag_name, template_name):
...
                                        'teachers_digital_platform.tool-urls')),
 
-    flagged_url('TDP_RELEASE',
+    flagged_url('TDP_CRTOOL',
             r'^practitioner-resources/youth-financial-education/curriculum-review/before-you-begin/',  # noqa: E501
             include_if_app_enabled('teachers_digital_platform',
                                     'teachers_digital_platform.begin-urls')),
 
-    flagged_url('TDP_RELEASE',
+    flagged_url('TDP_CRTOOL_PROTOTYPES',
```
Comment (Scotchester) [inline cfgov/cfgov/urls.py]: Final question: Do you want to go ahead and set up the flagged URL for the search interface, so you don't have to make a PR for that later?

Comment (atuggle) [reply cfgov/cfgov/urls.py]: That is a great question.  Yes, yes I do want to go ahead and create that now :-)... just a moment

**Comment by atuggle** [comment]
@Scotchester I added the Search Interface flagged url, thanks for the suggestion!

**Pull request merged**